superset/common/query_context_processor.py

```diff
@@ -420,21 +487,19 @@ def processing_time_offsets(  # pylint: disable=too-many-locals,too-many-stateme
                         )
                     )
 
+                # modifies temporal column using offset
                 offset_metrics_df[index] = offset_metrics_df[index] - DateOffset(
                     **normalize_time_delta(offset)
                 )
```

Comment:
the index of `offset_metrics_df` seems constructing an appropriate index for the result of dataframe, if the align of the results incorrect, I think we should find root cause from here.

Reply:
I'm not sure if I understand what you're suggesting. Can you explain more?